rts/motoko-rts/src/types.rs

```diff
@@ -452,6 +453,22 @@ impl Array {
     }
 }
 
+#[rustfmt::skip]
+#[repr(C)] // See the note at the beginning of this module
+pub struct Region {
+    pub header: Obj,
+    pub id: u16,
+    pub page_count: u64,
+    pub vec_ptr: *mut Blob, // Blob knows its own capacity
+}
+
+impl Region {
+    pub unsafe fn payload_addr(self: *const Self) -> *mut Value {
+        self.offset(1) as *mut Value // skip array header
+    }
+    // to do
+}
+
```

Comment:
@crusso (cc @luc-blaeser @kentosugama)

Here is me copying that note from our meeting into the run time system.

It seems like `payload_addr` is part of every `impl` (getting the address beyond the header, I suppose), and that `.offset(1)` is something that comes from saying `#[repr(C)]`.  Interesting way of doing C programming in Rust.

I'll keep looking at `Array` and others to figure out any patterns.